src/patternfly/components/InputGroup/input-group.scss

```diff
@@ -40,41 +41,33 @@
   display: flex;
   width: 100%;
 
-  * {
-    border-radius: 0;
+  .pf-c-button {
+    position: relative;
 
+    &,
     &::after {
       border-radius: 0;
     }
-  }
 
-  .pf-c-button:first-child::after,
-  .pf-c-dropdown:first-child .pf-c-dropdown__toggle::before,
-  .pf-c-form-control:first-child,
-  &__text:first-child {
-    border-radius: var(--pf-c-input-group--BorderRadius) 0 0 var(--pf-c-input-group--BorderRadius);
-  }
+    &:first-child {
```

Comment:
How about a comment here clarifying what this selector is doing? (and the last-child one also)

Reply:
added